spec/models/manageiq/providers/kubernetes/container_manager/scanning/job_spec.rb

```diff
@@ -144,6 +144,20 @@ def fetch_metadata(*_args)
       end
     end
 
+    it 'should add correct environment variables' do
+      att_name = 'http_proxy'
+      my_value = "MY_TEST_VALUE"
+      @ems.custom_attributes.create(:section => described_class::ATTRIBUTE_SECTION,
+                                    :name    => att_name,
+                                    :value   => my_value)
+      allow_any_instance_of(described_class).to receive_messages(:kubernetes_client => MockKubeClient.new)
```

Comment:
Yes, agreed.  Removing `any_instance_of` will require changing the code under test so the test isn't so difficult to write.  That's for a followup PR.  But yes, it's [broken us before](https://github.com/ManageIQ/manageiq/issues/5814) and will do it again since rspec-mocks `any_instance_of` relies on brittle code based on ruby's internals.